egs2/TEMPLATE/asr1/pyscripts/utils/extract_xvectors.py

```diff
@@ -42,6 +44,82 @@ def get_parser():
+        self.toolkit = args.toolkit
+        self.device = device
+        from speechbrain.dataio.preprocess import AudioNormalizer
+
+        self.audio_norm = AudioNormalizer()
+        if self.toolkit == "speechbrain":
+            from speechbrain.pretrained import EncoderClassifier
+
+            self.model = EncoderClassifier.from_hparams(
+                source=args.pretrained_model, run_opts={"device": device}
+            )
+        elif self.toolkit == "rawnet":
+            if not os.path.exists("RawNet"):
+                git.Repo.clone_from("https://github.com/Jungjee/RawNet", "RawNet")
+            repo = git.Repo("RawNet")
+            repo.git.submodule("update", "--init", "--recursive")
+            sys.path.append("RawNet/python/RawNet3/models")
+            sys.path.append("RawNet/python/RawNet3")
+            subprocess.check_call(
+                [sys.executable, "-m", "pip", "install", "asteroid-filterbanks==0.4.0"]
+            )
+            from RawNet3 import RawNet3
+            from RawNetBasicBlock import Bottle2neck
+
+            self.model = RawNet3(
+                Bottle2neck,
+                model_scale=8,
+                context=True,
+                summed=True,
+                encoder_type="ECA",
+                nOut=256,
+                out_bn=False,
+                sinc_stride=10,
+                log_sinc=True,
+                norm_sinc="mean",
+                grad_mult=1,
+            )
+
+            self.model.load_state_dict(
+                torch.load(
+                    "RawNet/python/RawNet3/models/weights/model.pt",
+                    map_location=lambda storage, loc: storage,
+                )["model"]
+            )
+            self.model.to(device).eval()
+
+    def rawnet_extract_embd(self, audio, n_samples=48000, n_segments=10):
+        if len(audio.shape) > 1:
+            raise ValueError(
+                f"RawNet3 supports mono input only. Input data has a shape of {audio.shape}."
+            )
+        if len(audio) < n_samples:  # RawNet3 was trained using utterances of 3 seconds
+            shortage = n_samples - len(audio) + 1
+            audio = np.pad(audio, (0, shortage), "wrap")
+        audios = []
+        startframe = np.linspace(0, len(audio) - n_samples, num=n_segments)
+        for asf in startframe:
+            audios.append(audio[int(asf) : int(asf) + n_samples])
+        audios = torch.from_numpy(np.stack(audios, axis=0).astype(np.float32)).to(
+            self.device
+        )
+        with torch.no_grad():
+            output = self.model(audios)
+        return output.mean(0).detach().cpu().numpy()
+
+    def __call__(self, wav, in_sr):
+        wav = self.audio_norm(torch.from_numpy(wav), in_sr).to(self.device)
+        if self.toolkit == "speechbrain":
+            embeds = self.model.encode_batch(wav).detach().cpu().numpy()[0]
+        elif self.toolkit == "rawnet":
+            embeds = self.rawnet_extract_embd(wav.detach().cpu().numpy())
```

Comment:
Since RawNet inputs raw waveforms directly, I recommend to use the same process for loading the utterance. 
In RawNet repository, I directly put waveforms loaded with soundfile without `self.audio_norm`

Reply:
Thanks. I had added `self.audio_norm` because RawNet only accepts 16kHz-sampled waveforms and TTS often uses the higher sampling rates. For clarification, I have replaced it with `librosa.resample`. If you have any other suggestions, please let me know.

Reply:
Great! [96b8a71](https://github.com/espnet/espnet/pull/4884/commits/96b8a715b31c47abbfb0f9662452391e6c5142ab) looks fine.